### PR TITLE
Contributing.md

### LINKED ISSUE #191: Update CONTRIBUTING.md
See examples: https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors

For the moment can be a simple file with minimal guidelines; not sure if `nox` options instructions should be stored in this file (as it is now) or somewhere else, but it is helpful to have them.

### PR DESCRIPTION
**Pull request by alepev**
Closes #191 

Includes conventions discussed in https://github.com/thehyve/delphyne-template/pull/73

Related to https://github.com/thehyve/delphyne-template/pull/76

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,23 +1,194 @@
...
+  nox -s build
+  ```
+  
+### Creating pull requests <a name="pull-requests"></a>
+
+Once your changes are ready to be submitted, push them to your fork of the GitHub repo before creating a pull request. 
+Depending on the type of issue the pull request is resolving, it should merge onto the appropriate branch:
+
+- pull requests to delphyne: merge to `release/X.X.X` branch if specified in the issue milestone, otherwise `master`.
+- pull request to delphyne-template: merge to `release/X.X.X` branch if specified in the issue milestone, otherwise `dev`.
```
Comment (alepev) [inline CONTRIBUTING.md:146]: do we know how to use issue milestones? could be useful..

**Comment by alepev** [comment]
To see the result: https://github.com/thehyve/delphyne/blob/contributing/CONTRIBUTING.md

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,23 +1,194 @@
...
+
+## Creating project releases
+
+These instructions are intended for delphyne and delphyne-template maintainers.
+
+### Release conventions
+
+Both delphyne and delphyne-template make use of [semantic versioning](https://semver.org/).
+
+#### delphyne releases
```
Comment (Spayralbe) [inline CONTRIBUTING.md:161]: Mention somewhere here that a new release of delphyne should be reflected in the requirements.txt file of delphyne-template

**Review by Spayralbe — COMMENTED** [review]
Clear and thorough ✨

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,23 +1,194 @@
...
+It is your responsibility as a developer to maintain compatibility between the two projects,
+making sure that changes to one repository are reflected by complementary changes in the other.
+
+Note that the `requirements.txt` file of delphyne-template 
+only specifies a dependency to delphyne up to the minor version (e.g. 0.1), 
+meaning that delphyne-template should always remain compatible with any further micro releases of delphyne (e.g. 0.1.2).
+
+### Getting started
+
+To contribute to the project, you should be already familiar with [git](https://git-scm.com/).
```
[the remark this answers is not in the document]
Comment (alepev) [reply CONTRIBUTING.md:50]: agreed

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,23 +1,194 @@
...
+  Then you are welcome to create a new _Feature request_ issue 
+  [here](https://github.com/thehyve/delphyne/issues/new?assignees=&labels=enhancement&template=feature_request.md&title=).
+
+Note that we might move issues to discussions and vice versa, whichever we find more appropriate.
+
+## Contributing as a developer
+
+You should always work on delphyne and [delphyne-template](https://github.com/thehyve/delphyne-template) side-by-side.
+It is your responsibility as a developer to maintain compatibility between the two projects,
+making sure that changes to one repository are reflected by complementary changes in the other.
```
[the remark this answers is not in the document]
Comment (alepev) [reply CONTRIBUTING.md:42]: Yes it came across as a bit harsh 😅 like "don't mess with my repo you punks!"

File: CONTRIBUTING.md (hunk 2 above)
Comment (alepev) [reply CONTRIBUTING.md:165]: done

**Comment by alepev** [comment]
- addressed comments
- changed titles & text to make clear the instructions are not just for delphyne, but also delphyne-template
- fixed couple links
- other

**Pull request merged**